tools/cli/domainCommands.go

```diff
@@ -854,6 +911,107 @@ func (d *domainCLIImpl) migrationDomainWorkFlowCheck(c *cli.Context) DomainMigra
 	}
 }
 
+func (d *domainCLIImpl) searchAttributesChecker(c *cli.Context) DomainMigrationRow {
+	ctx, cancel := newContext(c)
+	defer cancel()
+
+	// getting user provided search attributes
+	searchAttributes := c.StringSlice(FlagSearchAttribute)
+	if len(searchAttributes) == 0 {
+		return DomainMigrationRow{
+			ValidationCheck:  "Search Attributes Check",
+			ValidationResult: true,
+		}
+	}
+
+	// Parse the provided search attributes into a map[string]IndexValueType
+	requiredAttributes := make(map[string]types.IndexedValueType)
+	for _, attr := range searchAttributes {
+		parts := strings.SplitN(attr, ":", 2)
+		if len(parts) != 2 {
+			ErrorAndExit(fmt.Sprintf("Invalid search attribute format: %s", attr), nil)
+		}
+		key, valueType := parts[0], parts[1]
+		ivt, err := parseIndexedValueType(valueType)
+		if err != nil {
+			ErrorAndExit(fmt.Sprintf("Invalid search attribute type for %s: %s", key, valueType), err)
+		}
+		requiredAttributes[key] = ivt
+	}
+
+	d.frontendClient = initializeFrontendClient(c)
```

Comment:
this should not been initialized here. We need to:
1. change newDomainCLI func to ensure frontend client is always initialized
2. remove all initializations in all checkers.